frontend/src/components/core/EnterNickname.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import ErrorMessage from './Error';
+import Loading from './Loading';
+import { LargeText, Text } from './Text';
+import { LargeCenterInputText, LargeInputButton } from './Input';
+import { ErrorResponse } from '../../api/Error';
+import { isValidNickname } from '../../api/Socket';
+
+// The type of enter nickname page (create or join).
+export const ENTER_NICKNAME_PAGE = Object.freeze({
+  CREATE: 'create',
+  JOIN: 'join',
+});
```

Comment:
I made this initially because I thought I could use this for if statements if there were different design changes on create vs. join pages; however, I think you're right that this would just be better as a parameter.

Fixed!